ui/app/serializers/clients/activity.js

```diff
@@ -1,6 +1,58 @@
 import ApplicationSerializer from '../application';
 import { formatISO } from 'date-fns';
 
+export default ApplicationSerializer.extend({
+  flattenDataset(payload) {
```

Comment:
`payload` has a specific connotation in serializer context (in my mind anyway) so it's surprising to see us iterating over it below, because I expect it to be a certain shape. I'd call this something specific related to what you actually expect it to be, like `byNamespaceArray`